utils.go

```
@@ -2,13 +2,32 @@
+	}
+	return 0
+}
+
+func nonzero16(x uint16) uint16 {
+	if x != 0 {
+		return 1
+	}
+	return 0
+}
+
+func zero32(x uint32) uint32 {
```

golangci-lint (1.64.8)

func `zero32` is unused

(unused)